leetcodePython/SlidingWindow/slidingwindow_2781.py

```diff
@@ -0,0 +1,14 @@
+from typing import List
+
+
+class Solution:
+    def longestValidSubstring(self, word: str, forbidden: List[str]) -> int:
+        forbidden_set = set(forbidden)
+        left_bound = 0
+        result = 0
+        for right in range (len(word)):
+            for left in range (max(left_bound,right-10),right+1):
+                if word[left:right+1] in forbidden_set:
+                    left_bound = left+1
```

Comment:
Since we start and shrink window from left to right. 
* Whenever we find the string exist in forbidden, the only think we can do is to keep shrinking the window
* The time we keep shrinking the window, we might find it match other forbidden, so we have to keep shrinking
* Every time we find a match, we shrink the actual window left bound.
* After we shrink iterate the whole window. We know the window size we left for this iteration
* After every single iteration, we know the maximum window